### PR TITLE
Correctly set AvFlags if NtlmChallenge contains MIC

### PR DESCRIPTION
**Pull request by hierynomus**

### CODE REVIEW

File: src/main/java/com/hierynomus/ntlm/functions/NtlmV2Functions.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * [MS-NLMP].pdf 2.2.2.7 NTLM v2: NTLMv2_CLIENT_CHALLENGE
+     * <p>
+     * 3.3.2 NTLM v2 Authentication
+     * Set temp to ConcatenationOf(Responserversion, HiResponserversion, Z(6), Time,
+     * ClientChallenge, Z(4), ServerName, Z(4))
+     *
+     * @param targetInformation
```
[the remark this answers is not in the document]
Comment (ecki) [reply src/main/java/com/hierynomus/ntlm/functions/NtlmV2Functions.java:82]: @sonatype-lift showing one more line context would be cool ,)

File: src/main/java/com/hierynomus/ntlm/functions/NtlmV1Functions.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.Arrays;
+import java.util.Random;
+
+import com.hierynomus.ntlm.NtlmException;
+import com.hierynomus.security.Cipher;
+import com.hierynomus.security.SecurityException;
+import com.hierynomus.security.SecurityProvider;
+
+class NtlmV1Functions {
+    static final byte[] LMOWFv1_SECRET = new byte[] { 0x4B, 0x47, 0x53, 0x21, 0x40, 0x23, 0x24, 0x25 }; // KGS!@#$%
```
[the remark this answers is not in the document]
Comment (hierynomus) [reply src/main/java/com/hierynomus/ntlm/functions/NtlmV1Functions.java:13]: @sonatype-lift ignore

File: src/main/java/com/hierynomus/smbj/auth/NtlmAuthenticator.java [hunk 3]
```diff
@@ -61,102 +91,166 @@ public NtlmAuthenticator create() {
...
+            return response;
+        }
+
+        // Ensure we set TARGET_INFO
+        negotiateFlags.add(NTLMSSP_NEGOTIATE_TARGET_INFO);
+        TargetInfo clientTargetInfo = createClientTargetInfo(serverNtlmChallenge);
+
+        long time = FileTime.now().getWindowsTimeStamp();
+        if (clientTargetInfo.hasAvPair(AvId.MsvAvTimestamp)) {
+            time = ((AvPairTimestamp) clientTargetInfo.getAvPair(AvId.MsvAvTimestamp)).getValue().getWindowsTimeStamp();
```
[the remark this answers is not in the document]
Comment (hierynomus) [reply src/main/java/com/hierynomus/smbj/auth/NtlmAuthenticator.java:183]: @sonatype-lift ignore

File: src/main/java/com/hierynomus/mssmb2/SMB2CompressionTransformHeader.java [hunk 4]
```diff
@@ -25,7 +25,7 @@
  * [MS-SMB2] 2.2.42 SMB2 COMPRESSION_TRANSFORM_HEADER
  * <p>
  * The SMB2 COMPRESSION_TRANSFORM_HEADER is used by the client or server when sending compressed messages.
- * This optional header is only valid for the SMB 3.1.1 dialect<73>.
+ * This optional header is only valid for the SMB 3.1.1 dialect&lt;73&gt;.
  */
 public class SMB2CompressionTransformHeader implements SMBHeader {
     public static final byte[] COMPRESSED_PROTOCOL_ID = {(byte) 0xFC, 'S', 'M', 'B'};
```
[the remark this answers is not in the document]
Comment (hierynomus) [reply src/main/java/com/hierynomus/mssmb2/SMB2CompressionTransformHeader.java]: @sonatype-lift ignoreall

**Pull request merged**